core/src/main/java/org/springframework/security/authorization/method/PostAuthorizeAuthorizationManager.java

```diff
@@ -76,7 +81,13 @@ public AuthorizationDecision check(Supplier<Authentication> authentication, Meth
 				mi.getMethodInvocation());
 		this.expressionHandler.setReturnObject(mi.getResult(), ctx);
 		boolean granted = ExpressionUtils.evaluateAsBoolean(attribute.getExpression(), ctx);
-		return new AuthorizationDecision(granted);
+		AuthorizationDecision decision = new AuthorizationDecision(granted);
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug(LogMessage.format(
+					"Authorizing method post-invocation '%s' with attribute '%s', authentication name '%s' and decision '%s'",
+					mi, attribute, authentication.get().getName(), decision));
```

Comment:
`getName()` could be PII, like maybe an email address.

You could log `authentication.get()` more safely since an application can override the `toString()` method, but I think it may be best to leave it out of the message altogether. Leaving it out also means that we don't activate a supplier unnecessarily, which is better for performance if that means a session lookup.